cnd/src/start_swap.rs

```diff
@@ -1,31 +1,41 @@
 use crate::{
-    db::{ForSwap, Save, Swap},
+    db::{ForSwap, Save},
     halight, hbit, herc20,
-    network::{comit::RemoteData, WhatAliceLearnedFromBob, WhatBobLearnedFromAlice},
-    protocol_spawner::{ProtocolSpawner, Spawn},
-    storage::{Load, Storage},
-    LocalSwapId, Protocol, Role, Side,
+    network::{RemoteData, WhatAliceLearnedFromBob, WhatBobLearnedFromAlice},
+    Load, LocalSwapId, Protocol, ProtocolSpawner, Role, Side, Spawn, Storage,
 };
-use chrono::offset::Utc;
+use chrono::NaiveDateTime;
 
-pub async fn start_swap(
+#[derive(Clone, Copy, Debug)]
+pub struct DecisionSwap {
+    pub id: LocalSwapId,
+    pub role: Role,
+    pub alpha: Protocol,
+    pub beta: Protocol,
+}
+
+#[derive(Clone, Copy, Debug)]
+pub struct Swap<A, B> {
+    pub role: Role,
+    pub alpha: A,
+    pub beta: B,
+    pub start_of_swap: NaiveDateTime,
+}
+
+pub async fn save_and_start_swap(
```

Comment:
I think this function should be part of the `network` module and only `start_swap` should be outside.

Reason being that `RemoteData` is a network type so this will only called from within the `network` module anyway.

Reply:
https://github.com/comit-network/comit-rs/pull/2796